packages/lit-dev-content/site/docs/ssr/client-api.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Lit SSR client API
+eleventyNavigation:
+  key: Client API
+  parent: Server-side rendering
+  order: 3
+---
+
+## Lit Templates
+"Hydration" is the process of having Lit re-associate the expressions of a Lit template with the nodes they should update in the DOM. In order to "hydrate" Lit templates, the `hydrate` method from the `experimental-hydrate` module is provided in the `lit` package. Prior to updating a server-rendered container using `render`, you should first call `hydrate` on that container using the same template and data that was used to render on the server:
+
+```js
+import {render} from 'lit';
+import {hydrate} from 'lit/experimental-hydrate.js';
+import {myTemplate} from './my-template.js';
+// Initial hydration required before render:
+// (must be same data used to render on the server)
+const initialData = getInitialAppData();
+hydrate(myTemplate(initialData), document.body);
+
+// After hydration, render will efficiently update the server-rendered DOM:
+const update = (data) => render(myTemplate(data), document.body);
+```
+
+## Lit Elements
+When LitElements are server rendered, their shadow root contents are emitted inside a `<template shadowroot>`, also known as a [Declarative Shadow Root](https://web.dev/declarative-shadow-dom/), a new browser feature that is shipping in [Chrome](https://developer.chrome.com/blog/new-in-chrome-90/#declarative). Declarative shadow roots automatically attach their contents to a shadow root on the template's parent element when parsed. For browsers that do not yet implement declarative shadow root, there is a [`template-shadowroot` polyfill](https://github.com/webcomponents/template-shadowroot), described below.
+
+Because the `hydrate` function above does not descend into shadow roots, it only works on one scope of the DOM at a time. To hydrate `LitElement` shadow roots, load the `lit/experimental-hydrate-support.js` module, which installs support for `LitElement` automatically hydrating itself when it detects it was server-rendered with declarative shadow DOM. This module should be loaded before the `lit` module is loaded, to ensure hydration support is properly installed.
```

Comment:
Your reading is correct. I suppose it's unfortunate that the function for polyfilling for declarative shadow roots is named `hydrateShadowRoots`. I don't know if showing two different code snippet is advisable since the user likely should include the polyfill support for their users.

Also correct that this mistakenly does not include the `lit/experimental-hydrate-support.js`.

I do think it would be better to show the approach described here https://github.com/lit/lit/tree/main/packages/labs/eleventy-plugin-lit#example-bootup-strategy which is more recently written and efficient. Though I wonder if it's trying too much to be efficient with the preloading link tags and the dsd-pending styling is extra noise.